lib/mux/base.ex

```diff
@@ -43,7 +43,7 @@ defmodule Mux.Base do
   def new(token_id, token_secret, opts \\ []) do
     opts = Keyword.merge(@defaults, opts) |> Enum.into(%{})
 
-    Tesla.build_client([
+    Tesla.client([
```

Comment:
I believe this was intentionally left as `Tesla.build_client` for a specific reason:

This lib has a dependency on ` {:tesla, "~> 1.0"},`. Notice this semvar version will include everything above and including 1.0 and everything below 2.0. That includes `1.0` and `1.1`. The problem is, the earlier versions of Tesla do not define `Tesla.client` - they only define `Tesla.build_client`. Our options are:

1. Make Mux as widely compatible as possible wrt to `Tesla`. If someone wants to bring Tesla `1.0` or `1.1` to the party, then we will let them.
2. Make this code change, and update the dep to be `~> 1.2`. If someone's application is using an older version of Tesla they will not be able to use the Mux SDK until they upgrade to a newer Tesla version.

We weighed the trade-offs, and determined that (1) is the answer. The only downside is that developers with a newer Tesla version get a deprecation warning printed to the console.

For now, that still seems like the right decision, when there's a tradeoff like this, we typically want to air on the side of wide compatibility.

Reply:
This was absolutely an oversight on my part, I didn't realize that it would cause issues with early versions of Tesla. I will revert the commit and resolve this conversation.